query-engine/prisma/src/tests/test_api.rs

```diff
@@ -0,0 +1,269 @@
+use crate::{
+    context::PrismaContext,
+    request_handlers::{GraphQlBody, GraphQlRequestHandler, RequestHandler, SingleQuery},
+    PrismaRequest, PrismaResponse,
+};
+use migration_connector::*;
+use migration_core::{
+    api::{GenericApi, MigrationApi},
+    commands::{ApplyMigrationInput, InferMigrationStepsInput, ResetCommand},
+};
+use quaint::{
+    ast::*,
+    connector::ConnectionInfo,
+    visitor::{self, Visitor},
+};
+use sql_migration_connector::SqlMigrationConnector;
+use std::{collections::HashMap, sync::Arc};
+use test_setup::*;
+
+pub struct QueryEngine {
+    context: Arc<PrismaContext>,
+}
+
+impl QueryEngine {
+    pub async fn request(&self, body: impl Into<SingleQuery>) -> serde_json::Value {
+        let request = PrismaRequest {
+            body: GraphQlBody::Single(body.into()),
+            headers: HashMap::new(),
+            path: String::new(),
+        };
+
+        match GraphQlRequestHandler.handle(request, &self.context).await {
+            PrismaResponse::Single(response) => serde_json::to_value(response).unwrap(),
+            _ => unreachable!(),
+        }
+    }
+}
+
+pub struct TestApi {
+    connection_info: ConnectionInfo,
+    migration_api: MigrationApi<sql_migration_connector::SqlMigrationConnector, sql_migration_connector::SqlMigration>,
```

Comment:
We could make this more abstract so we don't have to have another `TestApi` for non-sql connectors when we get there, but for now it's good.

Reply:
Yikes.